### PR TITLE
Disable failed email receipts for Stripe Extension

### LINKED ISSUE #14715: [TTPReq] Disable failed receipt sending for Stripe extension

### PR DESCRIPTION
**Pull request by staskus**
Closes: #14715
Context: peNR48-13b-p2

## Description

Stripe Extension doesn't automatically set the order to failed after payment fails. Therefore, failed receipts are not automatically sent and receipts sent via the API after the payment don't have a failed template.

## Solution

Disable the new failed receipt functionality for Stripe Extension:

1. `isEligibleForSuccessfulPaymentEmailReceipts` returns true for 9.5 WooCommerce version
2. `isEligibleForFailedPaymentEmailReceipts` returns true for the current payment gateway being WooPayments, and matching   WooCommerce 9.5 and WooPaymnets 8.6 versions

## Steps to reproduce

I suggest using https://testpovilasstore.mystagingwebsite.com/wp-admin/ for testing, @iamgabrielma you should have access. It has both Stripe and WooPayments.

### WooPayments 8.6 + WooCommerce 9.5

1. Select WooPayments gateway
2. Make a payment with a failed amount (1.05)
3. Pay with Card Reader / Tap to Pay
4. Confirm that "Email Receipt" button appears
5. Send the email
6. Confirm the email is received

### Stripe Extension + WooCommerce 9.5

1. Select Stripe gateway
2. Make a payment with a failed amount (1.05)
3. Pay with Card Reader / Tap to Pay
4. Confirm that "Email Receipt" button doesn't appear

## Testing information

Additionally, I tested successful payment and attached the customer before and after the payment.

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/Order Details/Receipts/ReceiptEligibilityUseCase.swift [hunk 1]
```diff
@@ -44,25 +48,53 @@ final class ReceiptEligibilityUseCase: ReceiptEligibilityUseCaseProtocol {
...
+    ///
     func isEligibleForPointOfSaleReceipts(onCompletion: @escaping (Bool) -> Void) {
         guard featureFlagService.isFeatureFlagEnabled(.sendReceiptsForPointOfSale) else {
             onCompletion(false)
             return
         }
 
         Task { @MainActor in
-            async let isWooCommerceSupported = isPluginSupported(Constants.wcPluginName,
+            let isWooCommerceSupported = await isPluginSupported(Constants.wcPluginName,
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewModels/Order Details/Receipts/ReceiptEligibilityUseCase.swift:61]: Interesting one, I sort of thought that these were the same, but they aren't 🤔 
- `async let` creates the task to execute the function, and while will run async it does not await the result immediately, so it runs concurrently to other stuff we may have within the scope, and its non-blocking.
- await runs sequentially and it's blocking, as awaits immediately and must complete before moving to the next line.

**Review by iamgabrielma — APPROVED** [review]
Works as described :shipit: 

Edge case, in case we see something off when debugging: 

When I initially switched to Stripe the failed transaction still showed the email button when it shouldn't, the problem was Stripe having pending requirements but not showing a warning right away when switching payment gateways, so it just defaulted back to WCPay automatically when I've navigated to the Orders tab to process the order. 

This was resolved after going back to payment provider management and skipping the requirements: 

Simulator Screenshot - iPad mini (A17 Pro) - 2024-12-18 at 14 33 07

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/233">21.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/WooShipping Create Shipping Labels/WooShippingAddPackageViewModelTests.swift [hunk 2]
```diff
@@ -79,7 +79,7 @@ final class WooShippingAddPackageViewModelTests: XCTestCase {
                                                                      predefinedPackages: [predefinedPackage])]
                 let allPredefinedOptions = [WooShippingCarrierPredefinedOptions(carrierID: "usps",
                                                                                 predefinedOptions: predefinedOptions)]
-                completion(.success(WooShippingPackagesResponse(storeOptions: ShippingLabelStoreOptions.fake(),
+                completion(.success(WooShippingPackagesResponse(siteID: siteID,
```
Comment (staskus) [inline WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/WooShipping Create Shipping Labels/WooShippingAddPackageViewModelTests.swift]: @bozidarsevo there were compiler errors when building these tests on the trunk. I suppose some related changes got integrated at different times. I fixed it this way and tests succeeded.

Comment (bozidarsevo) [reply WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/WooShipping Create Shipping Labels/WooShippingAddPackageViewModelTests.swift]: Weird how https://github.com/woocommerce/woocommerce-ios/pull/14687/ did not fail with compiling :/

Comment (bozidarsevo) [reply WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/WooShipping Create Shipping Labels/WooShippingAddPackageViewModelTests.swift]: @rachelmcr made a separate PR with only this fix https://github.com/woocommerce/woocommerce-ios/pull/14722#pullrequestreview-2511633328

**Pull request merged**